### PR TITLE
Migrate stress cluster to use workload identity

### PR DESCRIPTION
**Pull request by benbp**
Notes TBD, a couple more edits need to be made (docs, test updates, and some more config deletion). Then I will publish out of draft.

### CODE REVIEW

File: eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1 [hunk 1]
```diff
@@ -216,6 +216,9 @@ function DeployStressPackage(
         Write-Host "Creating namespace $($pkg.Namespace) if it does not exist..."
         kubectl create namespace $pkg.Namespace --dry-run=client -o yaml | kubectl apply -f -
         if ($LASTEXITCODE) {exit $LASTEXITCODE}
+        # Give a few seconds for stress watcher to initialize the federated identity credential
+        # and create the service account before we reference it
+        Start-Sleep 5
```
Comment (benbp) [inline eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1:221]: I will update this to see if the namespace already exists and not sleep.

File: tools/stress-cluster/chaos/examples/network-stress-example/Chart.lock [hunk 2]
```diff
@@ -1,6 +1,6 @@
 dependencies:
 - name: stress-test-addons
-  repository: https://stresstestcharts.blob.core.windows.net/helm/
+  repository: file:///home/ben/azs/azure-sdk-tools/tools/stress-cluster/cluster/kubernetes/stress-test-addons
```
Comment (weshaggard) [inline tools/stress-cluster/chaos/examples/network-stress-example/Chart.lock:3]: I assume you plan to update this away from your local path.

File: tools/stress-cluster/cluster/kubernetes/stress-infrastructure/templates/stresswatcher.yaml [hunk 3]
```diff
@@ -27,12 +30,16 @@ spec:
       - name: stresswatcher
         imagePullPolicy: Always
         image: azsdkengsys.azurecr.io/stress/watcher:{{ .Values.tag }}
-        command: ["dotnet", "Stress.Watcher.dll"]
+        command: ["sh", "-c"]
+        args:
+          - >
+            az login --federated-token "$(cat $AZURE_FEDERATED_TOKEN_FILE)" --service-principal -u $AZURE_CLIENT_ID -t $AZURE_TENANT_ID;
```
Comment (weshaggard) [inline tools/stress-cluster/cluster/kubernetes/stress-infrastructure/templates/stresswatcher.yaml:36]: Is using a file the only real way to share this token?

**Review by weshaggard — APPROVED** [review]
Did a quick scan and things look reasonable.

File: tools/stress-cluster/cluster/kubernetes/stress-infrastructure/templates/stresswatcher.yaml (hunk 3 above)
Comment (benbp) [reply tools/stress-cluster/cluster/kubernetes/stress-infrastructure/templates/stresswatcher.yaml:36]: @weshaggard The file is the implementation method of AKS for federated auth, so unfortunately yes.

**Pull request merged**